numpy/testing/tests/test_utils.py

```diff
@@ -468,47 +468,70 @@ def test_report_fail_percentage(self):
         self.assertTrue("mismatch 25.0%" in msg)
 
 class TestArrayAlmostEqualNulp(unittest.TestCase):
-    @dec.knownfailureif(True, "Github issue #347")
-    def test_simple(self):
-        np.random.seed(12345)
-        for i in range(100):
-            dev = np.random.randn(10)
-            x = np.ones(10)
-            y = x + dev * np.finfo(np.float64).eps
-            assert_array_almost_equal_nulp(x, y, nulp=2 * np.max(dev))
-
-    def test_simple2(self):
-        x = np.random.randn(10)
-        y = 2 * x
-        def failure():
-            return assert_array_almost_equal_nulp(x, y,
-                                                  nulp=1000)
-        self.assertRaises(AssertionError, failure)
-
-    def test_big_float32(self):
-        x = (1e10 * np.random.randn(10)).astype(np.float32)
-        y = x + 1
-        assert_array_almost_equal_nulp(x, y, nulp=1000)
-
-    def test_big_float64(self):
-        x = 1e10 * np.random.randn(10)
-        y = x + 1
-        def failure():
-            assert_array_almost_equal_nulp(x, y, nulp=1000)
-        self.assertRaises(AssertionError, failure)
 
-    def test_complex(self):
-        x = np.random.randn(10) + 1j * np.random.randn(10)
-        y = x + 1
-        def failure():
-            assert_array_almost_equal_nulp(x, y, nulp=1000)
-        self.assertRaises(AssertionError, failure)
-
-    def test_complex2(self):
-        x = np.random.randn(10)
-        y = np.array(x, np.complex) + 1e-16 * np.random.randn(10)
+    def test_float64_pass(self):
+        nulp = 5
```

Comment:
I picked 5 because I wanted it to be a close to the precision limit, but I didn't want to be right at it it (nulp=1) in case that led to unforeseen problems. As it stands this is much tighter than the previous tests both in nulp and in the deviations needed to pass/fail the test. I can't think of a reason why we would need a loop. I'm happy to add one if you think it is needed.